packages/node_modules/cerebral/test/typescript/controller.ts

```diff
@@ -30,7 +30,10 @@ const controller: ControllerClass = Controller({
     module1,
     module2
   },
-  catch: new Map([
+  // We have to type map here or it will try to find a common
+  // interface for keys and none exists.
+  catch: new Map<CatchErrorConstructor, SignalChain>([
```

Comment:
@gaspard I suppose now we can remove type here. right?

Reply:
No, we cannot remove the type because TS will try to guess the common type of keys and will fail due to current TS limitations...

Reply:
![image](https://user-images.githubusercontent.com/21707/30912869-6e2c1ddc-a38e-11e7-8323-2e60ec803d0c.png)

Reply:
But maybe we can find a fix. Let me try.

Reply:
The issue is TS trying to find a common constructor signature and they are all different so it complains.

Reply:
@gaspard. hmm. but it works on my side.
![image](https://user-images.githubusercontent.com/36270/30914753-be96dc0a-a3a5-11e7-8f7d-dd4ec633e805.png)
![image](https://user-images.githubusercontent.com/36270/30914852-0783cb80-a3a6-11e7-8a5d-bd72ec82f786.png)

no complies, compiled without errors

Reply:
It works because you have only one key in the Map (HttpProviderError). As soon as you have more then one, TS tries to resolve the constructor signature and fails.